### PR TITLE
Add option to disable ctx providers

### PR DESCRIPTION
**Pull request by pkukielka**
## Test plan

N/A, unittests should pass.

When it will get adopted in JetBrains test plan will be as follow: 
https://github.com/sourcegraph/cody/pull/5384

### CODE REVIEW

File: lib/shared/src/mentions/api.ts [hunk 1]
```diff
@@ -51,21 +51,20 @@ export const SYMBOL_CONTEXT_MENTION_PROVIDER: ContextMentionProviderMetadata & {
     emptyLabel: 'No symbols found',
 }
 
-/** Metadata for all registered {@link ContextMentionProvider}s. */
-export function allMentionProvidersMetadata(): Observable<ContextMentionProviderMetadata[]> {
+const availableMentionProviders = [FILE_CONTEXT_MENTION_PROVIDER, SYMBOL_CONTEXT_MENTION_PROVIDER]
+export type OptionalProviderId = (typeof availableMentionProviders)[number]['id']
+
+export function mentionProvidersMetadata(
+    disabledMentionsProviders?: OptionalProviderId[] | undefined | null
```
Comment (dominiccooney) [inline lib/shared/src/mentions/api.ts:58]: Making this an object so the providers are labeled with `disable` will be more readable now and evolvable in future, something like:

```
mentionProvidersMetadata({disable: [SYMBOL_CONTEXT_MENTION_PROVIDER]})
```

File: lib/shared/src/mentions/api.ts [hunk 2]
```diff
@@ -51,21 +51,20 @@ export const SYMBOL_CONTEXT_MENTION_PROVIDER: ContextMentionProviderMetadata & {
...
+): Observable<ContextMentionProviderMetadata[]> {
     return openCtxMentionProviders().map(providers => [
-        FILE_CONTEXT_MENTION_PROVIDER,
-        SYMBOL_CONTEXT_MENTION_PROVIDER,
+        ...availableMentionProviders.filter(
+            provider => !disabledMentionsProviders?.includes(provider.id)
+        ),
         ...providers,
```
Comment (dominiccooney) [inline lib/shared/src/mentions/api.ts:64]: Why only filter the `availableMentionProviders`? Why not others?

**Review by dominiccooney — CHANGES_REQUESTED** [review]
A nice generalization of what web and JetBrains needs.

Let's filter all the providers, not just the two built-in ones? (Or make the API design a bit clearer...)

Let's add a unit test.

File: lib/shared/src/mentions/api.ts (hunk 2 above)
Comment (pkukielka) [reply lib/shared/src/mentions/api.ts]: Fixed

File: lib/shared/src/mentions/api.ts (hunk 1 above)
Comment (pkukielka) [reply lib/shared/src/mentions/api.ts]: Done

File: lib/shared/src/mentions/api.test.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import type { Observable } from 'observable-fns'
+import { describe, expect, it } from 'vitest'
+import {
+    FILE_CONTEXT_MENTION_PROVIDER,
+    SYMBOL_CONTEXT_MENTION_PROVIDER,
+    mentionProvidersMetadata,
+} from './api'
+
+function waitForObservable<T>(observable: Observable<T>): Promise<T> {
```
Comment (dominiccooney) [inline lib/shared/src/mentions/api.test.ts:9]: This already exists, it is `firstValueFrom`

File: lib/shared/src/mentions/api.test.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        const subscription = observable.subscribe(value => {
+            resolve(value)
+            subscription.unsubscribe()
+        })
+    })
+}
+
+describe('mentionProvidersMetadata', () => {
+    it('should return all providers when no options are provided', async () => {
+        const providers = await waitForObservable(mentionProvidersMetadata())
```
Comment (dominiccooney) [inline lib/shared/src/mentions/api.test.ts:20]:
```suggestion
        const providers = await firstValueFrom(mentionProvidersMetadata())
```

**Review by dominiccooney — APPROVED** [review]
Thank you for writing the test.

The test helper function can be de-duped with `firstValueFrom`, PTAL.

**Pull request merged**